### PR TITLE
[gis_web] Allow package:web 1.0.0

### PR DESCRIPTION
**Pull request by ditman**
Brackets the package:web dependency in gis_web to allow for `">=0.5.1 <2.0.0"`.

## Issues

This is needed so we can rebase and land:

* Part of: #7202 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by ditman** [comment]
(First pass, see what CI thinks...)

**Comment by ditman** [comment]
It's hard to test these, since they're a dependency only. CI seems happy, and I ran the tests as well. (Going to manually test google_sign_in_web real quick, which is not triggered in CI for this change)

**Review by kevmoo — APPROVED** [review]

**Comment by ditman** [comment]
@stuartmorgan I'm attempting to land a migration to `web:^1.0.0`. I think I can directly update most of the packages, but some dependencies of dependencies need to have a bracket in the web dependency.

I'm trying to get out of [this knot](https://github.com/flutter/packages/pull/7202#issuecomment-2246335991), and retry the big merge later.

File: packages/google_identity_services_web/pubspec.yaml [hunk 1]
```diff
@@ -2,14 +2,14 @@ name: google_identity_services_web
 description: A Dart JS-interop layer for Google Identity Services. Google's new sign-in SDK for Web that supports multiple types of credentials.
 repository: https://github.com/flutter/packages/tree/main/packages/google_identity_services_web
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+google_identiy_services_web%22
-version: 0.3.1+1
```
Comment (stuartmorgan) [inline packages/google_identity_services_web/pubspec.yaml:5]: Optional nit: This seems like a pretty trivial change; i would version it as `+2` instead.

**Review by stuartmorgan — APPROVED** [review]
> (Let me know if you have a faster/better/safer idea to do this!)

This should be entirely safe; I see in CI logs that `analyze` is indeed resolving to `web` 1.0.0, and `downgrade analyze` is still resolving to `0.5.1`, so we have analysis coverage of the whole range.

Ranges are totally fine, and in some cases preferable to straight upgrades.

File: packages/google_identity_services_web/pubspec.yaml (hunk 1 above)
Comment (ditman) [reply packages/google_identity_services_web/pubspec.yaml:5]: Rolled to +2!

**Pull request merged**